### ISSUE TITLE
Debugger crashing during launch on OSX in master branch

### ISSUE THREAD

**Issue by gregg-miskelly** [dotnet/coreclr#23773]
## Repro steps
On macOS, using current master bits (I used 3.0.100-preview4-011077), create a new console app (`dotnet new console`), and attempt to debug it.

Result:
Crash in the following code:
```
  * frame #0: 0x000000019c7d4eb3 libmscordaccore.dylib`CorUnix::InternalCreateEvent(CorUnix::CPalThread*, _SECURITY_ATTRIBUTES*, int, int, char16_t const*, void**) + 115
    frame #1: 0x000000019c7d4ff4 libmscordaccore.dylib`CreateEventW + 84
    frame #2: 0x000000019c408051 libmscordbi.dylib`CordbRCEventThread::Init() + 33
    frame #3: 0x000000019c41cd0a libmscordbi.dylib`Cordb::Initialize() + 106
    frame #4: 0x000000019bdea629 libvsdebugeng.impl.dylib`ManagedDM::CCoreClrAttachRequest::AttachToCoreClrProcessImpl(this=0x0000000100d180d0, pCorDebug=0x0000000100b30638) + 953 at CoreClrAttachRequest.h:155
    frame #5: 0x000000019bddaa0e libvsdebugeng.impl.dylib`ManagedDM::CCoreClrAttachRequest::AttachToCoreClrProcess(pCordb=0x0000000100b30638, pArg=0x0000000100d180d0, hrRuntimeLoaded=0) + 286 at CoreClrAttachRequest.h:87
    frame #6: 0x000000019c2cf438 libdbgshim.dylib`RuntimeStartupHelper::InvokeStartupCallback(char*, void*) + 712
    frame #7: 0x000000019c329355 libdbgshim.dylib`PAL_RuntimeStartupHelper::InvokeStartupCallback() + 229
    frame #8: 0x000000019c32910c libdbgshim.dylib`PAL_RuntimeStartupHelper::StartupHelperThread() + 140
    frame #9: 0x000000019c32904e libdbgshim.dylib`StartupHelperThread(void*) + 14
    frame #10: 0x000000019c32a5f8 libdbgshim.dylib`CorUnix::CPalThread::ThreadEntry(void*) + 328
    frame #11: 0x00007fff5f2ea6c1 libsystem_pthread.dylib`_pthread_body + 340
    frame #12: 0x00007fff5f2ea56d libsystem_pthread.dylib`_pthread_start + 377
    frame #13: 0x00007fff5f2e9c5d libsystem_pthread.dylib`thread_start + 13
```

**Comment by RussKeldorph** [comment]
@dotnet/dotnet-diag

**Comment by hoyosjs** [comment]
Fixed by #23924 and #23937. Should be included in the official preview 4 build.
